File: webforj-engine/src/main/java/com/webforj/component/html/elements/Strong.java
package com.webforj.component.html.elements;

import com.webforj.component.Component;
import com.webforj.component.element.annotation.NodeName;
import com.webforj.component.html.HtmlComponentContainer;

/**
 * Component representing a {@code strong} element.
 *
 * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong">Html strong
 *      Tag</a>
 *
 * @author Hyyan Abo Fakher
 * @since 23.06
 */
// We're purposefully ignoring the deep inheritance warning here because we've designed our class
// hierarchy to meet the unique requirements of our UI framework. This design closely aligns with
// our framework's specific goals and emphasizes the need for caution when considering any changes.
//
// Any changes to the inheritance structure should be thoughtfully evaluated in the context of our
// framework's needs. The current structure is essential for meeting those needs.
@SuppressWarnings("squid:S110")
@NodeName("strong")
public class Strong extends HtmlComponentContainer<Strong> {

  /**
   * Creates a new empty strong.
   */
  public Strong() {
    super();
  }

  /**
   * Creates a new strong with the given text.
   *
   * @param text the text
   */
  public Strong(String text) {
    super();
    setText(text);
  }

  /**
   * Creates a new strong with the given child components.
   *
   * @param components the child components
   */
  public Strong(Component... components) {
    super(components);
  }
}

File: webforj-kotlin/src/main/kotlin/com/webforj/kotlin/dsl/component/html/elements/Strong.kt
package com.webforj.kotlin.dsl.component.html.elements

import com.webforj.component.html.elements.Strong
import com.webforj.concern.HasComponents
import com.webforj.kotlin.dsl.WebforjDsl
import com.webforj.kotlin.dsl.init

/**
 * Creates an `Strong` with an optional [text] value.
 * ```
 * ... {
 *  Strong() // Empty Strong element
 *  Strong("text") // Strong element with text
 * }
 * ```
 * @param text Optional text to add to the `Strong`.
 * @param block The initialization steps of the `Strong`.
 * @return The configured `Strong` instance.
 * @see Strong
 * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong">Html strong
 *      Tag</a>
 */
fun @WebforjDsl HasComponents.strong(text: String? = null, block: @WebforjDsl Strong.() -> Unit = {}): Strong {
    val strong = text?.let { Strong(text) } ?: Strong()
    return init(strong, block)
}
